lib/api/api.js

```diff
@@ -606,6 +606,38 @@ api.prototype.sendTransfer = function(seed, depth, minWeightMagnitude, transfers
     })
 }
 
+/* --------------------------------------------------------------------------*/
+/**
+* Promotes a transaction by adding spam on top of it.
+* Will promote {maximum} transfers on top of the current one with {delay} interval.
+*
+* @Param {string} tail
+* @param {int} depth
+* @param {int} minWeightMagnitude
+* @param {array} transfer
+* @param {int} delay
+* @param {int} maximum
+* @param callback
+*
+    * @returns the output of the final sendTransfer call
+*/
+    /* ----------------------------------------------------------------------------*/
+api.prototype.promoteTransaction = function(tail, depth, minWeightMagnitude, transfer, delay, maximum, callback) {
+    var self = this;
+    if (!inputValidator.isHash(tail)) {
+        return callback(errors.invalidTrytes());
+    }
+    self.sendTransfer(transfer[0].address, depth, minWeightMagnitude, transfers, {reference: tail}, function(err, res) {
+        if (err == null && delay > 0 && maximum > 0) {
```

Comment:
maximum is off by one.
for example: if you call w/ `maximum=1`, `sendTransfer()` would be called once, then enter the recursion, called again & finally stopped.